helix-term/src/commands.rs

```diff
@@ -5516,14 +5519,41 @@ fn shrink_selection(cx: &mut Context) {
         // if not previous selection, shrink to first child
         if let Some(syntax) = doc.syntax() {
             let text = doc.text().slice(..);
-            let selection = object::shrink_selection(syntax, text, current_selection);
+            let selection = object::shrink_selection(syntax, text, current_selection.clone());
             doc.set_selection(view.id, selection);
         }
     };
     motion(cx.editor);
     cx.editor.last_motion = Some(Motion(Box::new(motion)));
 }
 
+fn select_sibling_impl<F>(cx: &mut Context, sibling_fn: &'static F)
+where
+    F: Fn(Node) -> Option<Node>,
+{
+    let motion = |editor: &mut Editor| {
+        let (view, doc) = current!(editor);
+
+        if let Some(syntax) = doc.syntax() {
+            let text = doc.text().slice(..);
+            let current_selection = doc.selection(view.id);
+            let selection =
+                object::select_sibling(syntax, text, current_selection.clone(), sibling_fn);
+            doc.set_selection(view.id, selection);
+        }
+    };
+    motion(cx.editor);
+    cx.editor.last_motion = Some(Motion(Box::new(motion)));
+}
+
+fn select_next_sibling(cx: &mut Context) {
+    select_sibling_impl(cx, &|node| Node::next_sibling(&node))
```

Comment:
```suggestion
    select_sibling_impl(cx, Node::next_sibling)
```

Does this compile?

Reply:
looks like the typing is slightly off, rustc wants a `&Node::next_sibling`:

```
error[E0308]: mismatched types
    --> helix-term/src/commands.rs:5550:29
     |
5550 |     select_sibling_impl(cx, Node::next_sibling)
     |                             ^^^^^^^^^^^^^^^^^^
     |                             |
     |                             expected reference, found fn item
     |                             help: consider borrowing here: `&Node::next_sibling`
     |
     = note: expected reference `&'static _`
                  found fn item `for<'r> fn(&'r helix_core::tree_sitter::Node<'_>) -> std::option::Option<helix_core::tree_sitter::Node<'_>> {helix_core::tree_sitter::Node::<'_>::next_sibling}`

error[E0308]: mismatched types
    --> helix-term/src/commands.rs:5554:29
     |
5554 |     select_sibling_impl(cx, Node::prev_sibling)
     |                             ^^^^^^^^^^^^^^^^^^
     |                             |
     |                             expected reference, found fn item
     |                             help: consider borrowing here: `&Node::prev_sibling`
     |
     = note: expected reference `&'static _`
                  found fn item `for<'r> fn(&'r helix_core::tree_sitter::Node<'_>) -> std::option::Option<helix_core::tree_sitter::Node<'_>> {helix_core::tree_sitter::Node::<'_>::prev_sibling}`

For more information about this error, try `rustc --explain E0308`.
error: could not compile `helix-term` due to 2 previous errors
```

But then doing `&Node::next_sibling`:

```
error[E0631]: type mismatch in function arguments
    --> helix-term/src/commands.rs:5550:29
     |
5550 |     select_sibling_impl(cx, &Node::next_sibling)
     |     -------------------     ^^^^^^^^^^^^^^^^^^^
     |     |                       |
     |     |                       expected signature of `for<'r> fn(helix_core::tree_sitter::Node<'r>) -> _`
     |     |                       found signature of `for<'r> fn(&'r helix_core::tree_sitter::Node<'_>) -> _`
     |     required by a bound introduced by this call
     |
note: required by a bound in `select_sibling_impl`
    --> helix-term/src/commands.rs:5532:8
     |
5530 | fn select_sibling_impl<F>(cx: &mut Context, sibling_fn: &'static F)
     |    ------------------- required by a bound in this
5531 | where
5532 |     F: Fn(Node) -> Option<Node>,
     |        ^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `select_sibling_impl`
```